src/object_view.hpp

```diff
@@ -81,14 +78,18 @@ class object_key {
     const detail::object *obj_{nullptr};
 };
 
-class object_view {
+class object_view final : public readable_object<object_view> {
```

Comment:
this is basically a read-only borrowed object right? So can't you move this to object.hpp and derive this for borrowed_object?

Reply:
Somewhat, deriving it from borrowed object would make it writable and I'd have to make some methods private. In addition, I think borrowed objects will need to have the allocator as well, once I introduce support, so they won't be exactly equivalent and deriving one from the other won't make sense anymore.

As with refactoring things further and moving this type to `object.hpp`, I left a TODO about it somewhere in `object.hpp` so I'll definitely be doing that at some point.